modules/core/arrow-core-data/src/main/kotlin/arrow/typeclasses/MonadPlus.kt

```diff
@@ -0,0 +1,52 @@
+package arrow.typeclasses
+
+import arrow.Kind
+
+/**
+ *  MonadPlus is a typeclass that extends a Monad by supporting choice and failure also.
```

Comment:
```suggestion
 *  MonadPlus is a typeclass that extends a Monad by supporting choice and failure.
 *  It is equal to [Alternative] in its api, but provides additional laws for how `flatMap` and `empty` interact.
```
Something like this might be a bit more descriptive